src/create-subscription-orders.js

```diff
@@ -0,0 +1,243 @@
+    await _buildQueryForTemplateOrdersThatNeedSubscriptionOrders(stateIds)
+
+  for await (const templateOrders of ctpClient.fetchPagesGraphQl(orderQuery))
+    await pMap(templateOrders, _processTemplateOrder, { concurrency: 3 })
+
+  return stats
+}
+
+async function _processTemplateOrder({
+  id,
+  version,
+  orderNumber,
+  custom: { customFieldsRaw },
+  state: { key: stateKey },
+}) {
+  const config = getSubscriptionConfig()
+  const headers = _buildHeaders(config)
+  const subscriptionOrderCreationUrl = config.subscriptionOrderCreationUrl
+
+  const deliveryDate = customFieldsRaw.find(
+    (attr) => attr.name === 'nextDeliveryDate'
+  ).value
+  const subscriptionOrderExists =
+    await _fetchOrderBySubscriptionTemplateOrderRefAndDeliveryDate(
+      id,
+      deliveryDate
+    )
+
+  if (subscriptionOrderExists) {
+    logger.info(
+      `Template order ${orderNumber}: ` +
+        `Subscription order was already created for date ${deliveryDate}. Will update the template order only.`
+    )
+    stats.subscriptionOrderAlreadyCreated++
+    await _updateTemplateOrder(id, version, customFieldsRaw, stateKey)
+  } else {
+    const response = await fetch(subscriptionOrderCreationUrl, {
+      method: 'POST',
+      headers,
+      body: JSON.stringify({ templateOrderId: id }),
+    })
+    if (_isError(response))
+      if (_isErrorRecoverable(response)) {
+        logger.error(
+          `Template order ${orderNumber}: ` +
+            `Recoverable error received when calling ${subscriptionOrderCreationUrl}. Will be retried next run. ` +
+            `Response status code ${response.status}`
+        )
+        stats.recoverableErrorUrlCalls++
+      } else {
+        logger.error(
+          `Template order ${orderNumber}: ` +
+            `Unrecoverable error received when calling ${subscriptionOrderCreationUrl}. ` +
+            'Please check the order and set its state back to "Active" ' +
+            `Response status code ${response.status}`
+        )
+        await updateOrderWithRetry(
+          apiRoot,
+          id,
+          [
+            {
+              action: 'transitionState',
+              state: {
+                typeId: 'state',
+                key: ERROR_STATE,
+              },
+            },
+          ],
+          version
+        )
+        stats.unrecoverableErrorUrlCalls++
+      }
+    else {
+      await _updateTemplateOrder(id, version, customFieldsRaw, stateKey)
+      stats.successUrlCalls++
+    }
+  }
+  stats.processedTemplateOrders++
+}
+
+async function _updateTemplateOrder(id, version, customFieldsRaw, stateKey) {
+  const schedule = customFieldsRaw.find(
+    (attr) => attr.name === 'schedule'
+  ).value
+  const cronExpression = parser.parseExpression(schedule)
+  const nextDeliveryCronDate = cronExpression.next()
+  const nextDeliveryDate = nextDeliveryCronDate.toDate()
+  const nextDeliveryDateISOString = nextDeliveryDate.toISOString()
+  const updateActions = [
```

Comment:
should we consider cutoffDays here ? 🤔

Reply:
Nope, cutoffDays is only for the first order to prevent customer getting 2 subscriptions too close to each other. In the next subscriptions it won't make sense anymore since customer will get the subscriptions in a correct interval defined by the schedule.